recipes/dd-opentracing-cpp/all/conanfile.py

```diff
@@ -0,0 +1,88 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class DatadogOpenTracingConan(ConanFile):
+    name = "dd-opentracing-cpp"
+    description = "Monitoring service for cloud-scale applications based on OpenTracing "
+    license = "Apache 2.0 License"
+    topics = ("instrumentration", "monitoring", "security", "tracing")
+    homepage = "https://github.com/DataDog/dd-opentracing-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, 14)
+
+        self.options["msgpack"].cpp_api = True
+
+    def requirements(self):
```

Comment:
It's considered bad behavior to change/override options of the user.
If a user builds with `-o msgpack:cpp_api=False`, it would be surprising if conan would override this argument.
Often it's better to check an option and complain.
```suggestion
    def validate(self):
        if not self.options["msgpack"].cpp_api:
            raise ConanInvalidConfiguration("dd-opentracing-cpp requires msgpack:cpp_api=True")

    def requirements(self):
```

Reply:
I checked this and that options is not even needed, my bad.

Reply:
Even better :wink: